Add fzf wrapper (#137)
vrothberg/vgrep: README.md

## Files
![](screenshots/vgrep-files.png)

## fzf

> 

Reviewer: Could you add some explanation here?  I have not heard of "fzf" until now and I guess there's an implicit assumption to copy the function below somewhere.   Having some guidance would be great.

Author: I've added some more description. If you would like to change anything I am open to it.

```shell
vgrep() {
  INITIAL_QUERY="$1"
  VGREP_PREFIX="vgrep --no-header "
  FZF_DEFAULT_COMMAND="$VGREP_PREFIX '$INITIAL_QUERY'" \
  | fzf --bind "change:reload:$VGREP_PREFIX {q} || true" --phony --tac --query "$INITIAL_QUERY" |
  | awk '{print $1}' | xargs -I{} -o vgrep --show {}

> }

Reviewer: Need to close the section with "```"

## Files
![](screenshots/vgrep-files.png)

## fzf

![](https://user-images.githubusercontent.com/7258858/103111382-b00c7f80-464c-11eb-9e47-c36ed89253a1.png)

> Running vgrep twice to open an editor is not ideal. The below function uses [fzf](https://github.com/junegunn/fzf) to iteratively search with vgrep and open your editor with one enter at the correct line.

Reviewer: I don't like the first sentence.  It really depends on the intended work flow.  We've discussed implementing a pager in vgrep in some issue, I believe in 2019, but I do not want to go down this road.  It's hard to get interactive UIs right and to fit the various work flows people have.  In this context, using `fzf` is a super cool way to get there.

I suggest to rewrite and explain what `fzf` is, and how it can enhance vgrep for the specific work flow.

Author: > I don't like the first sentence.

Do you have something in mind that you like? I have no idea.

> I suggest to rewrite and explain what fzf is

I think everyone should know fzf and if not I linked the repo where you can find out about it. I can't explain what fzf is in a few short sentences that are easy to understand because the tool is simply to powerful and does to much on its own.

> and how it can enhance vgrep for the specific work flow.

I explained that already.

```suggestion
Running vgrep twice to open an editor is not ideal. The below function uses [fzf](https://github.com/junegunn/fzf) to interatively search with vgrep and open your editor with one enter at the correct line.
```

Reviewer: > > I don't like the first sentence.
> 
> Do you have something in mind that you like? I have no idea.

Yes, as mentioned, I would prefer to explain what fzf is. "Not ideal" does not add information. "If you desire a more interactive experience than running `vgrep` twice to first search and another time to open an editor, you may have a look at `fzf`.", for example, may be more expressive.
 
> > I suggest to rewrite and explain what fzf is
> 
> I think everyone should know fzf and if not I linked the repo where you can find out about it. I can't explain what fzf is in a few short sentences that are easy to understand because the tool is simply to powerful and does to much on its own.

I have never heard of `fzf` until this pull request and do not assume that everyone does.  Linking to the GitHub page of `fzf` does not help me (at least) much as there's way too much information to understand how `fzf` and `vgrep` could interact.

Author: ```suggestion
If you desire a more interactive experience than running vgrep twice to first search and then to open an editor, you may have a look at fzf. The below function uses [fzf](https://github.com/junegunn/fzf) to iteratively search with vgrep and open your editor with one enter at the correct line.
```

Reviewer: s/interatively/interactively/

Besides that, LGTM. Thanks!

> To use it add the following function to your ``.bashrc`` and install fzf alongside vgrep and ripgrep.

Author: Not necessarily but normal grep is painfully slow and you do not want to wait a few seconds after typing.

> To use it add the following function to your ``.bashrc`` and install fzf alongside vgrep and ripgrep.

Reviewer: I think that's up to the user.  If `ripgrep` isn't a hard requirement, please remove it.

> To use it add the following function to your ``.bashrc`` and install fzf alongside vgrep and ripgrep.

Author: It is not per say a hard requirement but I imagine it being unusable without it and waiting several seconds after typing something would make this unusable.

```suggestion
To use it add the following function to your ``.bashrc`` and install fzf alongside vgrep and preferable ripgrep to have a smooth experience.
```
